WordPress/Classes/ViewRelated/Plugins/PluginDirectoryViewModel.swift

```diff
@@ -0,0 +1,144 @@
+            },
+            collectionCellSelected: { [weak presenter, weak self] plugin in
+                guard let capabilities = self?.siteCapabilities else { return }
+                presenter?.present(plugin: plugin, capabilities: capabilities)
+        })
+
+        let commonRowType = CollectionViewContainerRow<PluginDirectoryCollectionViewCell, PluginDirectoryEntry>.self
+
+        let configureCell: (PluginDirectoryCollectionViewCell, PluginDirectoryEntry) -> Void = { [weak self] cell, item in
+            let iconPlaceholder = Gridicon.iconOfType(.plugins, withSize: CGSize(width: 98, height: 98))
+            cell.logoImageView?.downloadImage(item.icon, placeholderImage: iconPlaceholder)
+            cell.authorLabel?.text = item.author
+            cell.nameLabel?.text = item.name
+
+            cell.accessoryView = self?.accessoryView(for: item)
+        }
+
+        let cellSelected: (PluginDirectoryEntry) -> Void = { [weak presenter] entry in
+            presenter?.present(directoryEntry: entry)
+        }
+
+        let featuredRow = commonRowType.init(data: featuredPlugins,
+                                             title: NSLocalizedString("Featured", comment: "Header of section in Plugin Directory showing featured plugins"),
+                                             secondaryTitle: nil,
+                                             action: nil,
+                                             configureCollectionCell: configureCell,
+                                             collectionCellSelected: cellSelected)
+
+        let popularRow = commonRowType.init(data: popularPlugins,
+                                            title: NSLocalizedString("Popular", comment: "Header of section in Plugin Directory showing popular plugins"),
+                                            secondaryTitle: NSLocalizedString("See All", comment: "Button in Plugin Directory letting users see more plugins"),
+                                            action: actionClosure(presenter: listPresenter, query: .feed(type: .popular)),
+                                            configureCollectionCell: configureCell,
+                                            collectionCellSelected: cellSelected)
+
+        let newRow = commonRowType.init(data: newPlugins,
+                                        title: NSLocalizedString("New", comment: "Header of section in Plugin Directory showing newest plugins"),
+                                        secondaryTitle: NSLocalizedString("See All", comment: "Button in Plugin Directory letting users see more plugins"),
+                                        action: actionClosure(presenter: listPresenter, query: .feed(type: .newest)),
+                                        configureCollectionCell: configureCell,
+                                        collectionCellSelected: cellSelected)
+
+        return ImmuTable(optionalSections: [
```

Comment:
I think you can just use `sections` here, not sure if there are any optionals involved